projects/addon-editor/extensions/media/utils.ts

```diff
@@ -0,0 +1,25 @@
+import {Attribute} from '@tiptap/core';
+import {Attrs, Node as NodeElement} from 'prosemirror-model';
+
+export function tuiExtractSources(node: NodeElement): Array<Array<Attrs | string>> {
+    const nodes =
+        ((node.content ?? {}) as unknown as {content: NodeElement[]})?.content ?? [];
+
+    return nodes.map(({type, attrs}) => [type.name, attrs]);
+}
+
+export function tuiParseMediaAttributes(): Record<string, Partial<Attribute>> {
+    return {
+        id: {parseHTML: element => element?.getAttribute(`id`)},
```

Comment:
Could be an array of attributes reduced to an object since it's all the same.

Reply:
fixed